indigo_api/models/tasks.py

```diff
@@ -280,6 +287,30 @@ def close(self, user, **kwargs):
         # send task_closed signal
         task_closed.send(sender=self.__class__, task=self)
 
+    # block
+    def may_block(self, view):
+        return view.request.user.is_authenticated and \
+               view.request.user.editor.has_country_permission(view.country) and \
+               view.request.user.has_perm('indigo_api.block_task')
+
+    @transition(field=state, source=['open'], target='blocked', permission=may_block)
+    def block(self, user, **kwargs):
+        if self.assigned_to:
+            self.assign_to(None, user)
+        blocked_by = kwargs['blocked_by'] or None
```

Comment:
Perhaps make this an explicit parameter with a default of None -- python will do the hard work for you.